spec/graphql/internal_representation/rewrite_spec.rb

```diff
@@ -124,7 +124,7 @@
 
       nut_selections = plant_selection.typed_children[schema.types["Nut"]]
       # `... on Tree`, `... on Nut`, and `NutFields`, but not `... on Fruit { ... on Tree }`
-      assert_equal 3, nut_selections["leafType"].ast_nodes.size
+      assert_equal 1, nut_selections["leafType"].ast_nodes.size
```

Comment:
On the one hand, this isn't the _intended_ result here, it's meant to keep track of all the AST nodes that were merged into this part of the query. 

On the other hand, I don't know of a real use case for it, and 1 node will probably be enough.  Hmm.